src/hooks/useCreateEventAttestation.ts

```diff
@@ -0,0 +1,102 @@
+import { useSigner } from "@/hooks/useSigner";
+import { encodePacked, keccak256, zeroAddress } from "viem";
+import { EAS, SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
+import { useCallback, useEffect, useState } from "react";
+import type { Address } from "viem";
+import {
+  CREATE_EVENT_SCHEMA,
+  CREATE_EVENT_SCHEMA_UID,
+} from "@/components/CreateEventSchemaButton";
+
+export interface Event {
+  owner: Address;
+  name: string;
+  briefDescription: string;
+  fullDescription: string;
+  startsAt: number;
+  endsAt: number;
+  imageUrl: string;
+}
+
+export const CREATE_EVENT_SCHEMA_ENCODER = new SchemaEncoder(
+  CREATE_EVENT_SCHEMA
+);
+
+export const useCreateEventAttestation = () => {
+  // Address only valid for Optimism Sepolia
+  const EAS_CONTRACT_ADDRESS = "0x4200000000000000000000000000000000000021";
+
+  const signer = useSigner();
+  const [eas, setEas] = useState<EAS | null>(null);
+
+  useEffect(() => {
+    if (signer) {
+      const newEas = new EAS(EAS_CONTRACT_ADDRESS);
+      newEas.connect(signer);
+      setEas(newEas);
+    }
+  }, [signer]);
```

Comment:
don't we already have this code in another hook? You could just use that hook here, no? 🤔